src/AgentFramework.Core.Handlers/Hosting/IssuerProvisioningHostedService.cs

```diff
@@ -1,64 +1,71 @@
-        }
         /// <inheritdoc />
         public IssuerProvisioningHostedService(
-            IProvisioningService provisioningService,
-            IServiceProvider serviceProvider,
-            Action<IServiceProvider, IssuerProvisioningConfiguration> configuration)
-            : this(provisioningService)
+            IProvisioningService provisioningService, 
+            IWalletService walletService, 
+            IWalletRecordService recordService, 
+            IOptions<AgentOptions> agentOptions, 
+            IOptions<WalletOptions> walletOptions) 
+            : base(
+                provisioningService, 
+                walletService, 
+                recordService, 
+                agentOptions, 
+                walletOptions)
         {
-            if (configuration != null)
-            {
-                configuration(serviceProvider, Configuration);
-            }
         }
 
         /// <inheritdoc />
-        public async Task StartAsync(CancellationToken cancellationToken)
+        public override async Task StartAsync(CancellationToken cancellationToken)
         {
             try
             {
-                await _provisioningService.ProvisionAgentAsync(Configuration);
-            }
-            catch (WalletExistsException)
-            {
-                // Wallet already exists, swallow exception
-            }
-            catch (AgentFrameworkException ex) when (ex.ErrorCode == ErrorCode.WalletAlreadyProvisioned)
-            {
-                // Wallet already provisioned
+                await ProvisionAsync(cancellationToken);
+
+                var wallet = await _walletService.GetWalletAsync(
+                        configuration: _walletOptions.WalletConfiguration,
+                        credentials: _walletOptions.WalletCredentials);
+
+                if (_agentOptions.AgentKeySeed == null)
+                {
+                    _agentOptions.AgentKeySeed = CryptoUtils.GetUniqueKey(32);
+                }
+
+                var issuer = await Did.CreateAndStoreMyDidAsync(
+                    wallet: wallet,
+                    didJson: new
+                    {
+                        did = _agentOptions.IssuerDid,
+                        seed = _agentOptions.IssuerKeySeed
+                    }.ToJson());
+
+                var record = await _provisioningService.GetProvisioningAsync(wallet);
+                record.IssuerSeed = _agentOptions.IssuerKeySeed;
+                record.IssuerDid = issuer.Did;
+                record.IssuerVerkey = issuer.VerKey;
+                //record.TailsBaseUri = TailsBaseUri?.ToString();
+
+                await _recordService.UpdateAsync(wallet, record);
             }
-            catch (WalletStorageException)
+            catch
             {
-                // Aggregate exception thrown when using custom wallets
-
-                // TODO: TM: add support to Indy SDK to expose exception types
+                // OK
             }
```

Comment:
Is there a plan to log or throw any exception here if provisioning fails?

Reply:
Both can happen. We need to itemize allowable exceptions, like 'WalletAlreadyExistsException'